### PR TITLE
[XPU] Make checkpoint_engine worker device-agnostic

### PR DESCRIPTION
**Pull request by siju-samuel**
> [!IMPORTANT]
> **Depends on https://github.com/MoonshotAI/checkpoint-engine/pull/96**
> (adds Intel XPU support to checkpoint-engine). That PR must merge first
> `pyproject_xpu.toml` tracks checkpoint-engine's upstream `main`, which only
> resolves to XPU-capable code once #96 lands. 

## Motivation

SGLang's checkpoint-engine integration enables fast in-place weight updates
for a running server (the RL / online-serving use case), but the worker
hardcoded `torch.cuda` for device UUID/id resolution. As a result the ZMQ
handshake with checkpoint-engine's `ParameterServer` only worked on CUDA.

This makes the worker device-agnostic so the same path works on Intel XPU,
and along the way fixes a latent NPU bug: the ParameterServer keys NPU
devices as `NPU-<uuid>`, but the old CUDA fallback would have keyed them as
`GPU-<uuid>` — a key that never resolves.

## Modifications

- `srt/checkpoint_engine/checkpoint_engine_worker.py`
  - `_accelerator_type()` now delegates to SGLang's device-agnostic
    `get_device()` instead of assuming CUDA; device-module lookups go through
    `get_device_module()` (removing the `getattr(torch, ...)` indirection).
  - `get_device_uuid()` mirrors `checkpoint_engine.ps::_get_physical_gpu_id`
    exactly: NPU → `NPU-<uuid>`, every other accelerator (CUDA, XPU) →
    `GPU-<uuid>`. This is the fix for the NPU key mismatch above.
- `python/pyproject_xpu.toml`
  - Add the `checkpoint-engine` optional dependency for the XPU build. XPU
    support is not yet in a released checkpoint-engine, so it tracks upstream
    source (`git+https://github.com/MoonshotAI/checkpoint-engine.git`) until a
    release ships it, mirroring how `sgl-kernel-xpu` is referenced in the same
    file.
- `test/registered/unit/checkpoint_engine/test_checkpoint_engine_worker.py`
  - CPU-mockable tests for cuda/xpu/npu key-format routing and the
    `AssertionError`→`ValueError` wrapping.
  - A hardware-gated XPU test asserting the worker's real device UUID equals
    the key the `ParameterServer` independently derives (the actual handshake
    contract).
  - Registered for both CPU (`base-a-test-cpu`) and XPU
    (`stage-b-test-1-gpu-xpu`) CI.

## Accuracy Tests

Not applicable — this changes device-identity resolution for the weight-update
handshake, not model forward/kernels. Model outputs are unaffected.

## Speed Tests and Profiling

Not applicable — no change to the inference hot path. `get_device()` /
`get_device_module()` are called only during the weight-update handshake, not
per token.

## Checklist

- [x] Format your code according to the Format code with pre-commit.
- [x] Add unit tests according to the Run and add unit tests.
- [ ] Update documentation according to Write documentations.
- [ ] Provide accuracy and speed benchmark results (N/A — see above).
- [x] Follow the SGLang code style guidance.

---
### CI States

Latest PR Test (Base): :hourglass_flowing_sand: [Run #34508326812](https://github.com/sgl-project/sglang/actions/runs/34508326812)
Latest PR Test (Extra): :x: [Run #34508326537](https://github.com/sgl-project/sglang/actions/runs/34508326537)
Latest PR Test (AMD ROCm 10): :x: [Run #34508327013](https://github.com/sgl-project/sglang/actions/runs/34508327013)

### CODE REVIEW

File: python/sglang/srt/checkpoint_engine/checkpoint_engine_worker.py [hunk 1]
```diff
@@ -33,6 +35,14 @@
 logger = logging.getLogger(__name__)
 
 
+def _accelerator_type() -> str:
```
Comment (polisettyvarma) [inline python/sglang/srt/checkpoint_engine/checkpoint_engine_worker.py]: why not get_device, directly ?

File: python/pyproject_xpu.toml [hunk 2]
```diff
@@ -72,6 +72,7 @@ dependencies = [
 ]
 
 [project.optional-dependencies]
+checkpoint-engine = ["checkpoint-engine @ git+https://github.com/MoonshotAI/checkpoint-engine.git"]
```
Comment (polisettyvarma) [inline python/pyproject_xpu.toml]: after PR merge and any release let's use/pin it like CUDA

File: python/sglang/srt/checkpoint_engine/checkpoint_engine_worker.py [hunk 3]
```diff
@@ -100,17 +102,24 @@ def __init__(self, model_runner):
...
 
     def get_device_uuid(self) -> str:
-        """Get the UUID of current device."""
-        # Get device UUID for current device
-        device_id = torch.cuda.current_device()
+        """Physical GPU id, matching checkpoint-engine's ParameterServer key.
+
+        Must equal ps.py::_get_physical_gpu_id for the ZMQ handshake to resolve:
+        NPU uses an ``NPU-<uuid>`` key, every other accelerator ``GPU-<uuid>``."""
+        if is_npu():
```
Comment (msinnha1) [inline python/sglang/srt/checkpoint_engine/checkpoint_engine_worker.py:109]: we may avoid this is_npu condition if we use "get_device_uuid", right? in place of device_module.get_device_properties(device_id).uuid
https://github.com/sgl-project/sglang/blob/0111b290312aa224962397db86c04fe112539fb2/python/sglang/srt/platforms/device_mixin.py#L186

File: test/registered/unit/checkpoint_engine/test_checkpoint_engine_worker.py [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+import torch
+
+from sglang.srt.utils import is_xpu
+from sglang.test.test_utils import CustomTestCase
+
+# checkpoint-engine is an optional extra (sglang[checkpoint-engine]) that CI does
+# not install, and the worker module raises ImportError at import time without it.
+# Probe first so this file stays importable: an unguarded import would fail
+# collection and take the whole file down rather than skipping.
+_HAS_CHECKPOINT_ENGINE = importlib.util.find_spec("checkpoint_engine") is not None
```
Comment (mingfeima) [inline test/registered/unit/checkpoint_engine/test_checkpoint_engine_worker.py:32]: checkpoint-engine is not installed by default. therefore this test file will always be skipped?

**Comment by mingfeima** [comment]
@siju-samuel please fix lint error

**Comment by mingfeima** [comment]
help rebased this pr, otherwise cuda ci will be gated (skipped due to the base is too old)

**Comment by mingfeima** [comment]
https://github.com/sgl-project/sglang/actions/runs/34298045011/job/102332177200?pr=32382 XPU CI regression found.
